player/react/src/lib/components/ui-time-picker.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { TimePickerSelect, TimePicker, SelectItem } from '@carbon/react';
+import { commonSlots, slotsDisabled } from '../common-slots';
+import { SendSignal } from '../types';
+
+export interface TimePickerState {
+	id: string | number;
+	codeContext?: {
+		name: string;
+	};
+	type: string;
+	placeholder: string;
+	disabled?: boolean;
+	invalid?: boolean;
+	invalidText?: string;
+	light?: boolean;
+	size?: string;
+	value?: string;
+	label?: string;
+	timezones?: [];
+	showTimePeriod?: boolean;
+	timePeriod?: [];
+}
+
+export const type = 'time-picker';
+
+export const signals = ['valueChange', 'click', 'timezoneValueChange', 'timePeriodValueChange'];
+
+export const slots = {
+	...commonSlots,
+	...slotsDisabled,
+	invalid: 'boolean',
+	isInvalid: (state: TimePickerState) => ({
```

Comment:
i think the naming of the setter is fine, i don't think other component (i,e. combobox) use this naming. `setInvalid` is fine but no need for unnecessary changes. 

In terms of the need of `setInvalid`, please check combobox as well because the setter is similar.